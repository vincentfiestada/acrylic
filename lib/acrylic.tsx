/**
 * Re-export all components in one place
 */
export { default as IProps } from "./components/IProps";
export { default as IEventProps } from "./components/IEventProps";

/**
 * Layout Components
 */
export { default as ILayoutProps } from "./components/layout/ILayoutProps";
export { default as Section } from "./components/layout/Section";
export { default as Column } from "./components/layout/Column";
export { default as Row } from "./components/layout/Row";
export { default as Alert } from "./components/layout/Alert";
export { default as Toolbar } from "./components/layout/Toolbar";

/**
 * Typography Components
 */
export { default as ITextProps } from "./components/typography/ITextProps";
export { default as Title } from "./components/typography/Title";
export { default as Text } from "./components/typography/Text";
export { default as Icon } from "./components/typography/Icon";

/**
 * Control Components
 */
export { default as Button } from "./components/controls/Button";
export { default as IconButton } from "./components/controls/IconButton";

/**
 * Variants
 */
export { default as Variant } from "./components/Variant";
