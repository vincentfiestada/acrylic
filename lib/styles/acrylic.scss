@import "normalize.css";
@import "typography";
@import "controls";
