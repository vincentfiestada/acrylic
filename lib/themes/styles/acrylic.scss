@import "normalize.css";
@import "typography";
@import "icons";
@import "layout";
@import "controls";
@import "progress";
@import "animation";
