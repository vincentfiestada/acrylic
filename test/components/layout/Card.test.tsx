import * as React from "react";
import { mount } from "enzyme";
import { Card, CardBody, CardImage, Title, Text } from "../../../lib/acrylic";

describe("<Card />", () => {
	it("Should match the latest snapshots", () => {
		expect(
			mount(
				<React.Fragment>
					<Card>
						<CardImage src="image" />
						<CardBody>
							<Title>Card</Title>
							<Text>This is a card.</Text>
						</CardBody>
					</Card>
				</React.Fragment>
			)
		).toMatchSnapshot();
	});
});
