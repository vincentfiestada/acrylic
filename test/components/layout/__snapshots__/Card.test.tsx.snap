// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Should match the latest snapshots 1`] = `
<Card>
  <Section
    classes={
      Array [
        "acr-card",
        undefined,
      ]
    }
    elevation={1}
  >
    <section
      className="acr-section acr-elevated-1 acr-card"
    >
      <CardImage
        src="image"
      >
        <img
          className="acr-card-img"
          src="image"
        />
      </CardImage>
      <CardBody>
        <Section
          classes={
            Array [
              "acr-card-body",
              undefined,
            ]
          }
        >
          <section
            className="acr-section acr-card-body"
          >
            <Title>
              <h6
                className="acr-title"
              >
                Card
              </h6>
            </Title>
            <Text>
              <p
                className="acr-text"
              >
                This is a card.
              </p>
            </Text>
          </section>
        </Section>
      </CardBody>
    </section>
  </Section>
</Card>
`;
